bot/mc.go

```diff
@@ -4,43 +4,116 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"time"
 	"venova/sshcmd"
 
 	"github.com/gorcon/rcon"
 )
 
-func restartMinecraft() {
+func manageMinecraftCmd(ctx CommandCtx) error {
+	msg := ctx.Message.Message
+	sess := ctx.Session
+	args := ctx.Args
+	actions := map[string]struct {
+		Message string
+		Command string
+	}{
+		"up":      {"The Minecraft Server has been brought up", "up -d"},
+		"down":    {"The Minecraft server has been brought down", "down"},
+		"restart": {"The Minecraft server has been restarted.", "restart"},
+	}
+	if action, exists := actions[args[0]]; exists {
+		if memberHasRole(msg.Member, mcRoleId) || msg.Author.ID == morthisId {
+			mcMsg, _ := sess.ChannelMessageSend(msg.ChannelID, "Attempting to modify the minecraft server...")
+
+			go func() {
+				err := execDockerCompose(action.Command)
+				if err != nil {
+					log.Printf("%s", err)
+					ctx.Reply(fmt.Sprintf("%s", err))
+					return
+				}
+				time.Sleep(time.Second * 5)
+				sess.ChannelMessageEdit(msg.ChannelID, mcMsg.ID, action.Message)
+			}()
+			return nil
+		}
+	} else {
+		ctx.Reply("Action is not available")
+		return nil
+	}
```

Comment:
You should get into the habit of identifying and cleaning up if statements like this. Whenever you see a huge `if` block followed by a very small else that breaks control-flow (return/break/continue) somehow, that should probably be reworked to have the else statement checked before. In this example it's much cleaner (and safer when we touch it in the future) to write this as:

```suggestion
	action, exists := actions[args[0]]
	if !exists {
		ctx.Reply("Action is not available")
		return nil
	}
	if memberHasRole(msg.Member, mcRoleId) || msg.Author.ID == morthisId {
		mcMsg, _ := sess.ChannelMessageSend(msg.ChannelID, "Attempting to modify the minecraft server...")

		go func() {
			err := execDockerCompose(action.Command)
			if err != nil {
				log.Printf("%s", err)
				ctx.Reply(fmt.Sprintf("%s", err))
				return
			}
			time.Sleep(time.Second * 5)
			sess.ChannelMessageEdit(msg.ChannelID, mcMsg.ID, action.Message)
		}()
		return nil
	}
```